### PR TITLE
feat: add `enumerator`

### LINKED ISSUE #4155: Add `enumerate` to collections
See https://github.com/flix/flix/issues/3950 for details.

What type classes? Foldables?

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #4155

I'm not sure the type alias should be there. Thoughts?

### CODE REVIEW

**Comment by magnus-madsen** [comment]
No, lets get rid of the  type alias.

**Comment by jaschdoc** [comment]
RFR

File: main/src/library/DelayList.flix [hunk 1]
```diff
@@ -1169,6 +1169,13 @@ namespace DelayList {
         };
         Iterator(done, next)
 
+    ///
+    /// Returns an iterator over `l` zipped with the indices of the elements,
+    /// i.e. an alias for `toIterator(r, l) |> Iterator.zipWithIndex`.
```
Comment (magnus-madsen) [inline main/src/library/DelayList.flix:1174]: Lets not reveal the implementation details. I think that's bad.

Comment (jaschdoc) [reply main/src/library/DelayList.flix:1174]: Alrighty :)

**Pull request merged**